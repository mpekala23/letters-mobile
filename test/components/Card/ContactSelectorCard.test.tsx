import React from 'react';
import { ContactSelectorCard } from '@components';
import { fireEvent, render, toJSON } from '@testing-library/react-native';

jest.mock('date-fns', () => ({
  format: () => 'Jul 12',
}));

const setup = (propOverrides = {}) => {
  const props = {
    firstName: 'Jane',
    lastName: 'Doe',
    letters: [
      {
        isDraft: true,
        recipientId: 8,
        content: 'Text',
        dateCreated: 'Jul 12',
      },
    ],
    onPress: jest.fn(),
    ...propOverrides,
  };
  return {
    ...render(<ContactSelectorCard {...props} />),
    props,
  };
};

describe('Contact Selector Card component', () => {
  it('should match snapshot', () => {
    const { container } = setup();
    const tree = toJSON(container);
    expect(tree).toMatchSnapshot();
  });

  it('should display first name', () => {
    const { getByText } = setup();
    expect(getByText('Jane')).toBeDefined();
  });

  it('should display profile picture', () => {
    const { getByTestId } = setup();
    expect(getByTestId('profilePicture')).toBeDefined();
  });

  it('should fire onPress() on press of contact card', () => {
    const { props, getByText } = setup();
    fireEvent.press(getByText('Jane'));
    expect(props.onPress).toHaveBeenCalledTimes(1);
  });

  // TO-DO: Write tests for correct # of letters, last heard, miles after API integration
});
