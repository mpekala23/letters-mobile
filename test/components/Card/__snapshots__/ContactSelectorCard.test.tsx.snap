// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact Selector Card component should match snapshot 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <TouchableOpacity
    activeOpacity={0.2}
    style={
      Array [
        Object {
          "backgroundColor": "white",
          "borderRadius": 6,
          "marginVertical": 6,
          "padding": 15,
          "width": "100%",
        },
        Object {
          "elevation": 4,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 4,
            "width": 0,
          },
          "shadowOpacity": 0.2,
          "shadowRadius": 2.62,
        },
        undefined,
      ]
    }
    testID="ContactSelectorCard"
  >
    <View
      style={
        Array [
          Object {
            "flex": 1,
            "flexDirection": "row",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "paddingTop": 12,
            },
          ]
        }
      >
        <View
          pointerEvents="none"
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#F4F4F4",
              "borderRadius": 40,
              "height": 80,
              "justifyContent": "center",
              "overflow": "hidden",
              "width": 80,
            }
          }
        >
          <TouchableOpacity
            activeOpacity={0.2}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#F4F4F4",
                "borderRadius": 40,
                "height": 80,
                "justifyContent": "center",
                "overflow": "hidden",
                "width": 80,
              }
            }
            testID="profilePicture"
          >
            <Text
              style={
                Array [
                  Object {
                    "fontFamily": "Poppins-Regular",
                  },
                  Object {
                    "color": "#6D6D6D",
                    "fontSize": 20,
                    "fontWeight": "500",
                  },
                ]
              }
            >
              JD
            </Text>
          </TouchableOpacity>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "paddingLeft": 18,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontFamily": "Poppins-Bold",
                "fontSize": 20,
              },
            ]
          }
        >
          Jane
        </Text>
        <Text
          style={
            Array [
              Object {
                "fontFamily": "Poppins-Regular",
              },
              Object {
                "color": "#6D6D6D",
              },
            ]
          }
        >
          <Text
            style={
              Object {
                "color": "#000",
              }
            }
          >
            üíå
          </Text>
           
          received
          :
           
          1
        </Text>
        <Text
          style={
            Array [
              Object {
                "fontFamily": "Poppins-Regular",
              },
              Object {
                "color": "#6D6D6D",
              },
            ]
          }
        >
          <Text
            style={
              Object {
                "color": "#000",
              }
            }
          >
            üìÜ
          </Text>
           
          last heard from you
          :
           
          Jul 27
        </Text>
        <Text
          style={
            Array [
              Object {
                "fontFamily": "Poppins-Regular",
              },
              Object {
                "color": "#6D6D6D",
                "paddingBottom": 4,
              },
            ]
          }
        >
          <Text
            style={
              Object {
                "color": "#000",
              }
            }
          >
            ‚úàÔ∏è
          </Text>
           
          letters traveled
          :
        </Text>
      </View>
    </View>
  </TouchableOpacity>
</View>
`;
